generator/src/lib.rs

```diff
@@ -751,38 +714,47 @@ fn name_to_tokens(type_name: &str) -> Ident {
     format_ident!("{}", new_name.as_str())
 }
 
-fn map_identifier_to_rust(ident: Ident) -> TokenTree {
-    match ident.to_string().as_str() {
-        "VK_MAKE_VERSION" => {
-            TokenTree::Group(Group::new(Delimiter::None, quote!(crate::vk::make_version)))
-        }
-        s => format_ident!("{}", constant_name(s)).into(),
-    }
-}
-
 /// Parse and yield a C expression that is valid to write in Rust
 /// Identifiers are replaced with their Rust vk equivalent.
 ///
 /// Examples:
-/// - `VK_MAKE_VERSION(1, 2, VK_HEADER_VERSION)` -> `crate::vk::make_version(1, 2, HEADER_VERSION)`
+/// - `VK_MAKE_VERSION(1, 2, VK_HEADER_VERSION)` -> `MAKE_VERSION(1, 2, HEADER_VERSION)`
 /// - `2*VK_UUID_SIZE` -> `2 * UUID_SIZE`
 fn convert_c_expression(c_expr: &str) -> TokenStream {
-    fn rewrite_token_stream(stream: TokenStream) -> TokenStream {
+    fn rewrite_token_stream(stream: TokenStream, depth: u32) -> TokenStream {
         stream
             .into_iter()
-            .map(|tt| match tt {
+            .enumerate()
+            .map(|(i, tt)| match tt {
                 TokenTree::Group(group) => TokenTree::Group(Group::new(
-                    group.delimiter(),
-                    rewrite_token_stream(group.stream()),
+                    // Discard outmost parenthesis
+                    if depth == 0 && i == 0 && group.delimiter() == Delimiter::Parenthesis {
+                        // TODO: Emit `Delimiter::Brace` instead, and remove them
+                        // where constants are emitted?
+                        Delimiter::None
+                    } else {
+                        group.delimiter()
+                    },
+                    rewrite_token_stream(group.stream(), depth + 1),
```

Comment:
This really is a gross hack to get rid of the outmost parenthesis in an expression to please the linter (because defines are generally of the form `#define FOO(x) ( ... the expression ... )` and we put that expression including parenthesis inside a function - that's already guarded by brackets).